### PR TITLE
WT-12353 Switch to Windows 2022 in Evergreen

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

**Comment by etienneptl** [comment]
There is one failure that needs to be fixed: [catch2-unittest-test](https://spruce.mongodb.com/task/wiredtiger_windows_64_catch2_unittest_test_patch_33a8e0f0d5e77bc433f1c3a7519745e55b2f8a99_65e6b4af2fbabeb02253b336_24_03_05_05_59_13?execution=0).

File: src/docs/build-windows.dox [hunk 1]
```diff
@@ -17,7 +17,7 @@ Now proceed with @ref windows_building
 
 Building WiredTiger on Windows requires
 <a href="https://cmake.org/">CMake</a> as well as the Microsoft Visual
-C++ compiler in Microsoft Visual Studio 2017 (or newer).
+C++ compiler in Microsoft Visual Studio 2012 (or newer).
```
Comment (korteland) [inline src/docs/build-windows.dox:20]: Typo: `2012` -> `2022`

File: test/evergreen.yml [hunk 2]
```diff
@@ -5519,9 +5519,9 @@ buildvariants:
     - name: configure-combinations
 
 - name: windows-64
```
Comment (korteland) [inline test/evergreen.yml:5521]: `name` worth changing too?

**Review by korteland — COMMENTED** [review]
I had a look at the linked patch build. We should take a look at the `Command line warning D9025 : overriding '/Ob0' with '/Ob1'` style warnings, since it probably means whatever flags we're using are out of date.

The catch2 error is more interesting - that looks like the compiler whinging about the standard library itself. Not sure what might be the problem there.

File: src/docs/build-windows.dox (hunk 1 above)
Comment (etienneptl) [reply src/docs/build-windows.dox]: 610e2078ddf0d21f4788ed7a75e152005269742c

File: test/evergreen.yml (hunk 2 above)
Comment (etienneptl) [reply test/evergreen.yml]: 7dbe5fd90d707dd8b5f1100812a052d879da96b8

**Review by korteland — APPROVED** [review]
LGTM!

**Review by lukech — APPROVED** [review]
LGTM, thanks.

**Pull request merged**